src/app/services/doc-appoinment-service/doctor-appointments.service.ts

```diff
@@ -0,0 +1,69 @@
+import {Injectable} from '@angular/core';
+import {FirestoreService} from '../../data/http/firestore.service';
+import {DoctorsAppointmentDTO} from '../../data/modelDTO/doctors-appointment-dto';
+import {first, map} from 'rxjs/operators';
+import {Observable} from 'rxjs';
+import {convertSnapshots} from '../../data/utils/firestore-utils.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DoctorAppointmentsService {
+  private DOCTOR_COLLECTION = 'doctors/';
+  private APPOINTMENT_COLLECTION = '/appointments';
+
+
+  constructor(private firestoreService: FirestoreService) {
+  }
+
+  getAllAppointments(doctorId: string): Observable<DoctorsAppointmentDTO[]> {
+    return this.firestoreService.getCollection(this.getAppointmentUrl(doctorId)).pipe(
+      map(snaps => convertSnapshots<DoctorsAppointmentDTO[]>(snaps)),
+      first()
+    );
+    // firebase uses websocket to transfer data and first closes that connection after first value was transmited - for multiple tryes we will use take method
+
+    // todo: subscribe in component
+  }
+
+  getAppointmentById(appointmentId: string, doctorId: string): Observable<DoctorsAppointmentDTO> {
+    return this.firestoreService.getDocById(this.getAppointmentUrl(doctorId), appointmentId);
+    // todo: subscribe in component
+  }
+
+  createAppointment(app: DoctorsAppointmentDTO, doctorId: string): void {
```

Comment:
I was thinking to we could check the animal id and create an appointment only for that particular animal

Reply:
add tuple with animal.name and animal.id